src/components/Personalization/handlers/createProcessPropositions.js

```diff
@@ -10,6 +10,13 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
+const executeSequentially = promiseFunctions =>
+  promiseFunctions.reduce((sequence, promiseFunction) => {
+    return sequence.then(result =>
+      promiseFunction().then(nextResult => [...result, nextResult])
+    );
+  }, Promise.resolve([]));
+
```

Comment:
`executeSequentially` acts similar to `Promise.all`.  But instead of running all promises simultaneously and waiting for them all to finish, each promise is resolved in sequence one after the next.

This is necessary when rendering visual propositions that build on top of each other.  Consider a campaign that adds a new hero element to the page.  During authoring the customer revisits the hero element and changes the image within it.  Then later updates the text.  These three changes will be captured a 3 separate proposition items.  Each one needs the one proior to it to be completed first.